Content.Shared/Directions/SharedDirectionExtensions.cs

```diff
@@ -42,6 +42,39 @@ public static IEnumerable<Direction> RandomDirections()
             }
         }
 
+        /// <summary>
+        ///     Gets random directions until none are left. Only
+        /// returns the four cardinal direction (north, east, south, west)
+        /// </summary>
+        /// <returns>An enumerable of the directions.</returns>
+        public static IEnumerable<Direction> RandomCardinalDirection()
+        {
+            var directions = new[]
+            {
+                Direction.North,
+                Direction.East,
+                Direction.South,
+                Direction.West,
+            };
+
+            var robustRandom = IoCManager.Resolve<IRobustRandom>();
+            var n = directions.Length;
+
+            while (n > 1)
+            {
+                n--;
+                var k = robustRandom.Next(n + 1);
+                var value = directions[k];
+                directions[k] = directions[n];
+                directions[n] = value;
+            }
+
+            foreach (var direction in directions)
+            {
+                yield return direction;
+            }
+        }
```

Comment:
Probably better to just use a struct enumerator and use the built-in shuffle on the array.

Also you ToArray this later which is a waste of allocs. You're probably better off on the server just using a Span<T> for it.

Reply:
Check if this is better now.